pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -1870,6 +1870,38 @@ func TestPerPodSchedulingMetrics(t *testing.T) {
 		t.Fatalf("Failed to pop a pod %v", err)
 	}
 	checkPerPodSchedulingMetrics("Attempt twice with update", t, pInfo, 2, timestamp)
+
+	// Case 4: A gated pod is created and scheduled after lifting gate. The queue operations are
+	//  Add gated pod -> check unschedulablePods -> lift gate & update pod -> Pop.
+	c = testingclock.NewFakeClock(timestamp)
+	// Create a queue with PreEnqueuePlugin
+	m := map[string][]framework.PreEnqueuePlugin{"": {&preEnqueuePlugin{allowlists: []string{"foo"}}}}
+	queue = NewTestQueue(ctx, newDefaultQueueSort(), WithClock(c), WithPreEnqueuePluginMap(m), WithPluginMetricsSamplePercent(0))
+
+	// Create a pod without PreEnqueuePlugin label.
+	gatedPod := st.MakePod().Name("gated-test-pod").Namespace("test-ns").UID("test-uid").Obj()
+	queue.Add(gatedPod)
+	// Check pod is added to the unschedulablePods queue.
+	if getUnschedulablePod(queue, gatedPod) != gatedPod {
+		t.Errorf("Pod %v was not found in the unschedulablePods.", gatedPod.Name)
+	}
+	// Override clock to get different InitialAttemptTimestamp
+	c.Step(1 * time.Minute)
+
+	// Update pod with the required label to get it out of unschedulablePods queue.
+	updateGatedPod := gatedPod.DeepCopy()
+	updateGatedPod.Labels = map[string]string{"foo": ""}
+	queue.Update(gatedPod, updateGatedPod)
+
+	pInfo, err = queue.Pop()
+	if err != nil {
+		t.Fatalf("Failed to pop a pod %v", err)
+	}
+
+	if pInfo.Timestamp == *pInfo.InitialAttemptTimestamp {
+		t.Errorf("[Gated Pod] Pod timesamp and initialAttemptTimestamp cannot be with same value when pod is gated, Timestamp %v, InitialAttemptTimestamp %v", pInfo.Timestamp, *pInfo.InitialAttemptTimestamp)
+	}
```

Comment:
this is an internal detail. The check below should be enough.

Reply:
Addressed.